### ISSUE TITLE
feat[dev]: ✨ implement A11y Re-Improvements to sd-radio-button

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#2059]
## Description
This tasks describes the Figma implementation for: #2048 

This task describes the Storybook implementation for the sd-radio-button component as part of the A11y re-improvements outlined in Epic #2048.

A11y improvements for sd-radio-button can be found here: [sd-radio-button Notes](https://www.notion.so/matuzo/sd-button-1baf3f6e1bc08198a2d6ddf4df8b1e21).

------------------------------------------

# Notes
- Disabled radio buttons: See [[sd-button](https://www.notion.so/sd-button-1baf3f6e1bc08198a2d6ddf4df8b1e21?pvs=21)](https://www.notion.so/sd-button-1baf3f6e1bc08198a2d6ddf4df8b1e21?pvs=21) . Everything I wrote there about disabling form elements applies here as well.

## Accessibility information
- Nesting other interactive elements like links inside labels should be avoided.

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

**Comment by smfonseca** [comment]
Updated a11y information in the overview pages [branch](https://github.com/solid-design-system/solid/pull/2024)
